src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -2340,8 +2317,90 @@ public void Cut()
                     IsInActivePrompt(caretPosition))
                 {
                     CutOrDeleteCurrentLine(isCut: true);
+                    Home(false);
+                }
+            }
+
+            /// <summary>Implements <see cref="IInteractiveWindowOperations.Delete"/>.</summary>
+            public bool Delete()
+            {
+                _historySearch = null;
+                bool handled = false;
+                if (!TextView.Selection.IsEmpty)
+                {
+                    if (CutOrDeleteSelection(isCut: false))
+                    {
+                        MoveCaretToClosestEditableBuffer();
+                        handled = true;
+                    }
+                }
+                else if (IsInActivePrompt(TextView.Caret.Position.BufferPosition))
+                {
                     MoveCaretToClosestEditableBuffer();
                 }
+                return handled;
+            }
+
+            /// <summary>Implements <see cref="IInteractiveWindowOperations2.DeleteLine"/>.</summary>
+            public void DeleteLine()
+            {
+                _historySearch = null;
+                CutOrDeleteLine(isCut: false);
+            }
+
+            /// <summary>Implements <see cref="IInteractiveWindowOperations2.CutLine"/>.</summary>
+            public void CutLine()
+            {
+                _historySearch = null;
+                CutOrDeleteLine(isCut: true);
+            }
+
+            /// <summary>Cut/Delete all selected lines, or the current line if no selection. </summary>                  
+            private void CutOrDeleteLine(bool isCut)
+            {
+                if (TextView.Selection.IsEmpty)
+                {
+                    var caret = TextView.Caret;
+                    var position = caret.Position.BufferPosition;
+                    if (MapToEditableBuffer(position) != null ||
+                        IsInActivePrompt(position))
+                    {
+                        CutOrDeleteCurrentLine(isCut); 
+                        Home(false);                
+                    }
+                }
+                else
+                {
+                    var selection = TextView.Selection;
+
+                    var anchor = selection.AnchorPoint;
+                    var active = selection.ActivePoint;
+
+                    ITextSnapshotLine startLine, endLine;
+                    int startCol, endCol;
+
+                    selection.AnchorPoint.GetLineAndColumn(out startLine, out startCol);
+                    selection.ActivePoint.GetLineAndColumn(out endLine, out endCol);
+
+                    if (startLine.LineNumber > endLine.LineNumber)
+                    {
+                        var temp = startLine;
+                        startLine = endLine;
+                        endLine = temp;
+                    }
+
```

Comment:
What happens when startCol > endCol but startLine is same as endLine? Do we need to swap the startCol and endCol in that case?